### ISSUE TITLE
glfwGetMonitors should return GLFWmonitor* const *

### ISSUE THREAD

**Issue by ghuser404** [glfw/glfw#1109]
IMO, `GLFWmonitor** glfwGetMonitors` should return `GLFWmonitor* const *`, because with `GLFWmonitor**` the library is exposing access to its internal pointers.

If we are to compare to `const GLFWvidmode* glfwGetVideoModes`, it doesn't expose access to internal data, and although the interface differs a bit (no **), it's const-correct nevertheless.

This changes API, so might be a good candidate for GLFW 4.0.

**Comment by elmindreda** [comment]
Yeah, that would be more correct, and I'm usually a stickler for const, but...

I'd seen enough even experienced programmers make mistakes with or at least being unsure about multi-level const that I thought enforcing this in application code would cause more hassle especially for beginners than leaving it writable would.

It's definitely something to revisit for 4.0, though.
